test/extensions/filters/http/proto_api_scrubber/filter_test.cc

```diff
@@ -581,15 +651,368 @@ TEST_F(ProtoApiScrubberScrubbingTest, ScrubRequestNestedField) {
+                        [type.googleapis.com/envoy.extensions.filters.http.proto_api_scrubber.v3.RemoveFieldAction] {}
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  )pb",
+                    kBookstoreDescriptorRelativePath);
+  reSetupFilter(kBookstoreDescriptorRelativePath);
+
+  auto req_headers = TestRequestHeaderMapImpl{{":method", "POST"},
+                                              {":path", "/bookstore.Bookstore/CreateShelf"},
+                                              {"content-type", "application/grpc"}};
+
+  EXPECT_CALL(mock_decoder_callbacks_, sendLocalReply(_, _, _, _, _)).Times(0);
+  EXPECT_EQ(Envoy::Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(req_headers, false));
 
+  // Verify data path is also fine.
+  CreateShelfRequest request = makeCreateShelfRequest();
+  Envoy::Buffer::InstancePtr request_data = Envoy::Grpc::Common::serializeToGrpcFrame(request);
+  EXPECT_EQ(Envoy::Http::FilterDataStatus::Continue, filter_->decodeData(*request_data, true));
+}
+
+// Tests that a request is allowed if no specific method-level rule is configured for the method.
+TEST_F(MethodLevelRestrictionTest, MethodAllowedNoRule) {
+  setupFilterConfig(R"pb(
+    restrictions: {
+      method_restrictions: {
+        key: "/bookstore.Bookstore/ListShelves"
+        value: {
+          # No method_restriction field
+        }
+      }
+    }
+  )pb",
+                    kBookstoreDescriptorRelativePath);
+  reSetupFilter(kBookstoreDescriptorRelativePath);
+
+  auto req_headers =
+      TestRequestHeaderMapImpl{{":method", "POST"},
+                               {":path", "/bookstore.Bookstore/CreateShelf"}, // Different method.
+                               {"content-type", "application/grpc"}};
+
+  EXPECT_CALL(mock_decoder_callbacks_, sendLocalReply(_, _, _, _, _)).Times(0);
+  EXPECT_EQ(Envoy::Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(req_headers, false));
+}
+
+// Tests the behavior when the matcher result is UnableToMatch (e.g., due to missing inputs).
+// Currently, this test demonstrates the fallback behavior with the standard CelMatcher.
+TEST_F(MethodLevelRestrictionTest, MethodAllowedOnMatcherInsufficientData) {
+  // NOTE: Accurately simulating an UnableToMatch state from the CelMatcher
+  // typically requires a CEL expression dependent on inputs not available
+  // during decodeHeaders (e.g., dynamic metadata).
+  // Testing the filter's reaction to a generic Matcher returning UnableToMatch
+  // would ideally involve injecting a mock Matcher. This is not easily done
+  // with the current ProtoApiScrubberFilterConfig structure, which builds the
+  // matcher internally.
+  // One could potentially create and register a custom Matcher extension
+  // specifically designed to return UnableToMatch for testing purposes,
+  // but that adds considerable complexity to this test file.
```

Comment:
Would it make sense for the configuration to be rejected if the matcher isn't a CEL matcher (because of the limitation of this implementation)?